### ISSUE TITLE
Reporting time to create/delete major cloud resources 

### ISSUE THREAD

**Issue by meteorfox** [GoogleCloudPlatform/PerfKitBenchmarker#1015]
AFAIK, currently PKB only report resource creation/deletion time if the benchmark supports it (i.e cluster_boot).

We know that one key difference that makes cloud benchmarking different from the traditional benchmarking is that the times to allocate, configure and delete resources are metrics of interests. As [hinted here in this slide from one of the tech talks](https://docs.google.com/presentation/d/146bK4EP59Dh2vQEjAeECSjp-IJD6eFVwPvXLSTtS9LY/edit#slide=id.g8436545c6_0_304) written by @voellm.

Today, some of these measurements can be gathered or inferred from the logs,  but I think they should be reported as part of the results summary for every provider that supports them.

I think for the most part we have converged to a certain set of resources, such as Network, Block Storage, Compute Instance (VM) and Firewall. Perhaps Object-Storage, Load Balancers, and others could become other kinds of resources in the future, but that's another issue. Starting to output resource creation/deletion metrics for these can be a start.

What do you think?

**Comment by cbhall** [comment]
I agree that resource creation/deletion times are of interest and should be reported.

Actually, there is a related issue I am concerned with, which is to making the times relatively accurate.  As an example, #982 and #997 represent a problem that needs to be fixed (I am seeing run failures with 1.5.0 because ssh returns 255 if a VM is not fully booted).  However, we need to be careful of the delay used between repeated ssh attempts - I think 5 seconds delay is too long and introduces too much inaccuracy in the reported boot times.

**Comment by meteorfox** [comment]
@cbhall I completely agree. For most of these cloud providers, boot times are in the order of 10s of seconds if not faster and 5 seconds is a large percentage of the time.

The new provider implementation for OpenStack #942 will poll every second , which is the default value used in  the retry decorator for the `WaitForBootCompletion` method. I'd love to hear your feedback on the new implementation, you could try it today, just clone it from my fork. :smile:

**Comment by cbhall** [comment]
@meteorfox I picked up the new provider implementation and tried it out.  It is much cleaner than the original implementation!  I had some comments which I put into #942.
